File: problems/002-two-sum/go/brute-force.go
package main

// Obvious approach: try every pair until one sums to the target. The first
// thing anyone writes, and the O(n^2) baseline that clean's single-pass hash
// map exists to beat. Allocates nothing, so O(1) space.
func bruteForce(c map[string]any) any {
	nums := c["nums"].([]any)
	target := c["target"].(float64)
	for i := 0; i < len(nums); i++ {
		for j := i + 1; j < len(nums); j++ {
			if nums[i].(float64)+nums[j].(float64) == target {
				return []int{i, j}
			}
		}
	}
	return []int{}
}

File: problems/002-two-sum/dart/brute-force.dart
// Obvious approach: try every pair until one sums to the target. The first thing
// anyone writes, and the O(n^2) baseline that clean's single-pass map exists to
// beat. Allocates nothing, so O(1) space.
dynamic solve(Map<String, dynamic> c) {
  final nums = (c['nums'] as List).cast<num>();
  final target = c['target'] as num;
  for (var i = 0; i < nums.length; i++) {
    for (var j = i + 1; j < nums.length; j++) {
      if (nums[i] + nums[j] == target) return [i, j];
    }
  }
  return [];
}
